pygsti/objects/forwardsim.py

```diff
@@ -187,8 +191,16 @@ def probs(self, circuit, outcomes=None, time=None):
         -------
         probs : OutcomeLabelDict
             A dictionary with keys equal to outcome labels and
-            values equal to probabilities.
+            values equal to probabilities. If no target outcomes provided,
+            only non-zero probabilities will be reported.
         """
+        if outcomes is None:
+            try:
+                # TODO: Patch ralloc up into probs so it can be passed here (and in create_layout)
```

Comment:
I'd like to discuss what you have in mind here re: "patch ralloc up..."

Reply:
Finished piping up resource_alloc to probs (which is just passed through to the COPALayout).